### PR TITLE
Split markdown into blocks so that checkboxes are singled out

### PR DESCRIPTION
**Pull request by newhinton**
This will seperate the markdown into different RemoteViewElements.

Each RemoteViewElement has a type and a text. The text corresponds to the raw, unrendered markdown, and the type will allow the result to be used more precisely.

I did this, so that i could implement checkboxes:

image

I develop this library-extension in tandem with the actual widget. So while this is not really a draft, i would not merge this yet until i got the actual feature to be reviewed on the app-side.

See: https://github.com/nextcloud/notes-android/pull/1912

### CODE REVIEW

File: markdown/src/main/java/it/niedermann/android/markdown/remoteviews/RemoteViewElement.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package it.niedermann.android.markdown.remoteviews
+
+import java.util.ArrayList
+
+class RemoteViewElement(
+    val type: Int,
```
Comment (stefan-niedermann) [inline markdown/src/main/java/it/niedermann/android/markdown/remoteviews/RemoteViewElement.kt:6]: This should be an `Enum`

File: markdown/src/main/java/it/niedermann/android/markdown/MarkdownUtil.java [hunk 2]
```diff
@@ -159,14 +226,21 @@ private static String runForEachCheckbox(@NonNull String markdownString, @NonNul
                 }
             }
             if (!isInFencedCodeBlock) {
-                if (lineStartsWithCheckbox(lines[i]) && lines[i].trim().length() > EListType.DASH.checkboxChecked.length()) {
```
Comment (stefan-niedermann) [inline markdown/src/main/java/it/niedermann/android/markdown/MarkdownUtil.java:162]: I wonder why I used to only consider `EListType.DASH`... This should maybe refactored to loop through the `EListType` enum.... however, nothing you will have to fix here :)

**Review by stefan-niedermann — CHANGES_REQUESTED** [review]
1. Please use Java only in this library. There are good reasons to use Kotlin, but this library module is Java only and I do not want to mix it up.
2. Please add at least one unit test for your method, better a few more with some edge cases (`null` or empty `content` parameter)-

**Comment by stefan-niedermann** [comment]
Merged in #245 (I also fixed the review items I made)
